sips/sip-013/sip-013-pox-upgrade.md

```diff
@@ -0,0 +1,373 @@
+the stacker has not delegated to an operator, and that their funds are
+currently stacked. The caller may supply a new reward address for the
+extension.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### 2. `delegate-stack-extend`
+
+This method allows operators to re-lock the funds of one of their
+delegation participants for additional cycles *before* that participants
+lock has expired. The target use case for this is to allow delegation
+operators to repeatedly stack on behalf of their users without cooldowns.
+
+This method checks that the delegator is still authorized on behalf of
+the given stacker (and remains authorized until the end of the
+locked-for period) and that the funds are currently stacked.
+
+Note that, just as with the existing `delegate-stack-stx` function,
+this method locks *but does not commit* the user's STX. The delegation
+operator must invoke `stack-aggregation-commit` to set a reward address
+for the locked funds.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### 3. `stack-increase`
+
+This method allows direct stackers to lock additional funds for
+the remaining cycles on their current lock.
+
+This method checks that the stacker is still allowed to stack, that
+the stacker has not delegated to an operator, that their funds are
+currently stacked, and that they have enough unlocked funds to cover
+the increase. The caller may *not* supply a new reward address for the
+increase.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the locked amount in the
+stacker's STX account to correspond to the increased amount.
+
+### 4. `delegate-stack-increase`
+
+This method allows operators to lock additional funds for one of their
+delegation participants for the remaining cycles on that user's
+current lock.
+
+This method checks that the delegator is still authorized on behalf of
+the given stacker (and remains authorized until the end of the
+locked-for period), that the increased amount remains less than the
+delegation's `amount-ustx` field, that the user is currently locked,
+and that the user has enough unlocked funds to cover the increase.
+
+Note that, just as with the existing `delegate-stack-stx` function,
+this method locks *but does not commit* the user's STX. The delegation
+operator must invoke `stack-aggregation-commit` to set a reward address
+for the newly locked funds.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the locked amount in the
+stacker's STX account to correspond to the increased amount.
+
+### 5. `stack-unlock`
```

Comment:
Does this method exist yet?  Also, I'm not sure this is a desirable feature (at least, not without a steep early-withdrawal penalty).  Part of the appeal of PoX's current behavior is that it compels people to hodl.  Also, the number of reward cycles the stacker locks for is a strong signal to miners and the ecosystem that this stacker has full confidence in PoX for the lock-up duration.  If we take that away, then it means that everyone is potentially just locking for the next reward cycle (since they can just exit afterwards), and there's not really a point to having a lock-up duration.

Reply:
I think this would be fine if you unlocked your tokens in-between when you locked them and when your first payout reward cycle begins, however (is that what this is for?).

Reply:
Looking at the example below, if the purpose of this method is to facilitate changing the PoX reward address, then why not just have a method to do that with the already-locked tokens?

Reply:
> Does this method exist yet?

Some methods of this SIP have been implemented in `next`, most haven't yet. But, I think as far as getting feedback on the SIP goes, we should ignore what has or hasn't been implemented: I think feedback on the SIP should inform what and how things get implemented, rather than the inverse.

> Part of the appeal of PoX's current behavior is that it compels people to hodl. Also, the number of reward cycles the stacker locks for is a strong signal to miners and the ecosystem that this stacker has full confidence in PoX for the lock-up duration. If we take that away, then it means that everyone is potentially just locking for the next reward cycle (since they can just exit afterwards), and there's not really a point to having a lock-up duration.

This is a very useful feature in and of itself -- obviously, it would break incentives to allow users to unlock during a reward cycle, but before one that hasn't occurred yet shouldn't alter incentives. Rather, _without_ this feature, PoX would encourage users that wish to have the freedom to unlock to only stack for one cycle at a time, continuously invoking `stack-extend` to get the same support, but with much more overhead (and stress about timing transactions before prepare phases).

Reply:
> Rather, without this feature, PoX would encourage users that wish to have the freedom to unlock to only stack for one cycle at a time, continuously invoking stack-extend to get the same support, but with much more overhead (and stress about timing transactions before prepare phases).

I think that's desirable, actually.  If you want the option to stack for a shorter amount of time, and want the freedom to decide whether or not to extend in each cycle, then the system *should* require you to send a transaction in each cycle to extend your lock.  Put another way, the system should offer you a better deal for locking up for longer, since you locking up your tokens provides a service to the ecosystem (i.e. you signal confidence to miners that the network will be stable during your lockup).  In general, we *want* people locking up their STX for as long as possible, since it makes liquid STX more valuable, which in turn makes it easier for miners to sell their newly-minted STX at a profit.